help_function/src/onedpl_test_group_load.cpp

```diff
@@ -0,0 +1,195 @@
+// ====------ onedpl_test_group_load.cpp------------ *- C++ -* ----===//
+
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//
+// ===----------------------------------------------------------------------===//
+#include <sycl/sycl.hpp>
+#include <dpct/dpct.hpp>
+#include <dpct/dpl_utils.hpp>
+#include <iostream>
+#include <oneapi/dpl/iterator>
+
+
+template<dpct::group::load_algorithm T>
+bool helper_validation_function(const int* ptr, const char * func_name){
+  if ( T == dpct::group::load_algorithm::BLOCK_LOAD_DIRECT)
+  {
+     for (int i = 0; i < 512; ++i) {
+      if (ptr[i] != i) {
+        std::cout << func_name << "_blocked" <<" failed\n";
+        std::ostream_iterator<int> Iter(std::cout, ", ");
+        std::copy(ptr, ptr + 512, Iter);
+        std::cout << std::endl;
+        return false;
+        }
+     }
+     std::cout << func_name << "_blocked" <<" pass\n";
+     return true;
+  }
+    
+  else{
+    int expected[512];
+    int num_threads = 128;
+    int items_per_thread = 4;
+    for (int i = 0;i < num_threads; ++i){
+        for(int j=0;j < items_per_thread; ++j){
+          expected[i * items_per_thread +j] = j * num_threads +i;  
+        }
+      }
+    for (int i = 0; i < 512; ++i) {
+        if (ptr[i] != expected[i]) {
+          std::cout << func_name << "_striped" <<" failed\n";
+          std::ostream_iterator<int> Iter(std::cout, ", ");
+          std::copy(ptr, ptr + 512, Iter);
+          std::cout << std::endl;
+          return false;
+        }
+      }
+  
+    std::cout << func_name << "_striped" <<" pass\n";
+    return true;
+  }
+  return false;

```

Comment:
This `return false` can never happen.  Either remove it or switch it to `true` and remove the `true` returns from within the branches.